### PR TITLE
Add unit tests and examples / doctests for nx.descendants_at_distance

### LINKED ISSUE #4957: descendants_at_distance appears to be untested
AFAICT `descendants_at_distance` is not explicitly tested anywhere. I was confused by this because the coverage report for the `breadth_first_search` module where the function is defined is at 98% (notably - the untested line is in `descendants_at_distance`), but I don't see any explicit calls to `descendants_at_distance` when grepping the test suite.

Adding some tests should help codify behavior and avoid issues like #4952 .

### PR DESCRIPTION
**Pull request by AndrewEckart**
Documented some examples and added a couple of unit tests for `nx.descendants_at_distance`. Also did some minor refactoring to eliminate an unreachable line. Increases module coverage to 100%.

Fixes #4957

### CODE REVIEW

File: networkx/algorithms/traversal/breadth_first_search.py [hunk 1]
```diff
@@ -383,28 +383,37 @@ def descendants_at_distance(G, source, distance):
     -------
     set()
         The descendants of `source` in `G` at the given `distance` from `source`
+
+    Examples
+    --------
+    >>> G = nx.path_graph(3)
```
Comment (dschult) [inline networkx/algorithms/traversal/breadth_first_search.py:389]: Maybe make this path length longer so it is clear what distance means in this context.

File: networkx/algorithms/traversal/breadth_first_search.py [hunk 2]
```diff
@@ -383,28 +383,37 @@ def descendants_at_distance(G, source, distance):
...
-        if current_distance == distance:
-            return queue
-
-        current_distance += 1
-
-        next_vertices = set()
-        for vertex in queue:
+    while current_distance < distance:
+        next_layer = set()
+        for vertex in current_layer:
```
Comment (dschult) [inline networkx/algorithms/traversal/breadth_first_search.py:411]: If you are going to rename and reorder anyway, could you change `vertex` to `node` everywhere?  It is the NetworkX standard word (along with edges) for how to describe the parts of a graph, and while we're here it'd be nice to bring this into that standard for this function.   :}

**Review by dschult — COMMENTED** [review]
Thanks for this!

File: networkx/algorithms/traversal/breadth_first_search.py (hunk 1 above)
Comment (AndrewEckart) [reply networkx/algorithms/traversal/breadth_first_search.py]: Enlarged it slightly and changed the source to a node which is not an endpoint of the path, but happy to change it further if needed.

File: networkx/algorithms/traversal/breadth_first_search.py (hunk 2 above)
Comment (AndrewEckart) [reply networkx/algorithms/traversal/breadth_first_search.py]: Done!

**Review by rossbar — APPROVED** [review]
Excellent, thanks @AndrewEckart !

**Pull request merged**